File: src/main/java/org/apache/prepbuddy/smoothingops/WeightedSlidingWindow.java
package org.apache.prepbuddy.smoothingops;

public class WeightedSlidingWindow extends SlidingWindow {
    private Weights weights;

    public WeightedSlidingWindow(int size, Weights weights) {
        super(size);
        this.weights = weights;
    }

    @Override
    public void add(double value) {
        if (isFull())
            queue.remove();
        int size = queue.size();
        Double weightValue = weights.get(size);
        queue.add(weightValue);
    }

    public Double average() {
        return sum();
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/smoothers/WeightedSlidingWindow.scala
package org.apache.datacommons.prepbuddy.smoothers

import org.apache.datacommons.prepbuddy.exceptions.{ApplicationException, ErrorMessages}

class WeightedSlidingWindow(windowSize: Int, weights: Weights) extends SlidingWindow(windowSize) {
    if (windowSize != weights.size){
        throw new ApplicationException(ErrorMessages.WINDOW_SIZE_AND_WEIGHTS_SIZE_NOT_MATCHING)
    }

    def average: Double = weights.multiplyWith(windowValues).sum
}
